src/lib/php/Application/ObligationCsvImport.php

```diff
@@ -181,34 +183,56 @@ private function handleCsvObligation($row)
   function AssociateWithLicenses($licList, $obPk, $candidate=False)
   {
     $associatedLicenses = "";
-    if ($candidate == False && empty($licList))
-    {
-      $return = "Obligation topic '$row[topic]' was added without being associated with any license.";
-    }
-    else
+    $message = "";
+
+    $licenses = explode(";",$licList);
+    foreach ($licenses as $license)
     {
-      $licenses = explode(";",$licList);
-      foreach ($licenses as $license)
+      $licId = $this->obligationMap->getIdFromShortname($license, $candidate);
+      if ($this->obligationMap->isLicenseAssociated($obPk, $licId, $candidate))
       {
-        $licId = $this->obligationMap->getIdFromShortname($license, $candidate);
-        if ($this->obligationMap->isLicenseAssociated($obPk, $licId, $candidate))
-        {
-          continue;
-        }
+        continue;
+      }
 
+      if (!empty($licId))
+      {
         $this->obligationMap->associateLicenseWithObligation($obPk, $licId, $candidate);
         if ($associatedLicenses == "")
         {
           $associatedLicenses = "$license";
+        } else {
+          $associatedLicenses .= ";$license";
         }
-        else
+      } else {
+        if ($message == "")
```

Comment:
@deveaud-m just a question: so in both cases the message is set to the same string, but one is appending and another is replacing (if the string is empty), concatenation could work on the empty string as well. Just asking if we could change this into a straight concatenation removing the conditional?

Reply:
@mcjaeger Sure, go ahead!